wfl/Models/Restaurant.cs

```diff
@@ -19,5 +21,22 @@ public bool CanHaveLinkToMenu()
             }
             return false;
         }
+        public static bool CanConstructFromDto(RestaurantDto dto)
+        {
+            if (dto?.Name == null || (dto.MenuLink != null && !Uri.IsWellFormedUriString(dto.MenuLink.ToString(), UriKind.Absolute)))
+            {
+                return false;
+            }
+            return true;
+        }
+        public static async Task<bool> CanUpdateRestaurantAsync(RestaurantDto dto, string id, IDBManager<Restaurant> restaurantDBManager)
```

Comment:
In that case, it isn't necessary to have this method at all. If the restaurant entity is not null, the same method to check if you can create a restaurant will allow you to check if you can update it as well.